helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -367,7 +590,22 @@ public Object removeRoutingTableChangeListener(
    */
   public Set<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName, String state,
       List<String> resourceTags) {
-    return _routingTableRef.get().getInstancesForResourceGroup(resourceGroupName, state,
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResourceGroup(resourceGroupName, state,
+        resourceTags);
+  }
+
+  public Set<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName, String state,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
Removed.